sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -39,12 +39,12 @@ public ManagedIdentityClient(CredentialPipeline pipeline, ResourceIdentifier res
 
         public ManagedIdentityClient(ManagedIdentityClientOptions options)
         {
+            _options = options;
```

Comment:
should the options be cloned rather than storing a reference to the one the user passed so they can't be mutated once the class is instantiated?

Reply:
I considered this, but our existing clone APIs don't seem to work here since we don't know what type the nested TokenCredentialOptions are. But perhaps we can make it work since the options we should have there would be limited

Reply:
actually - they are already guaranteed to be cloned [in the DAC path](https://github.com/Azure/azure-sdk-for-net/blob/e545aecefff9e8688cfe7eb26ae9001d86860d8e/sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs#L123) so we should be able to clone in the non DAC case.